### PR TITLE
Mark `blockifier` modified regions

### LINKED ISSUE #392: Document exactly what the cheated methods are changing
To make sure we are capable of keeping `Cheatnet` up to date we need to have a clear understanding of the modified foreign code.

### PR DESCRIPTION
**Pull request by Arcticae**
Closes #392

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/rpc.rs [hunk 1]
```diff
@@ -846,50 +854,76 @@ pub fn execute_constructor_entry_point(
...
+    call_to_external: bool,
+    entry_point_selector: EntryPointSelector,
+    calldata: Calldata,
+    remaining_gas: &mut u64,
+) -> SyscallResult<ReadOnlySegment> {
+    let entry_point_type = if call_to_external {
+        EntryPointType::External
+    } else {
+        EntryPointType::L1Handler
+    };
```
Comment (Arcticae) [inline crates/cheatnet/src/rpc.rs:894]: I've added this method so the code mirrors the blockifier code directly, so it can be easier to update @war-in

**Comment by MaksymilianDemitraszek** [comment]
I understand the intentions but I am not a big fan of this PR, imo it encourages blind copying of the blockifier code without understanding 😞.

**Comment by war-in** [comment]
> I get the intentions but I am not a big fan of this PR, imo it encourages developers to blind copy the blockifier code without understanding 😞.

I can't agree. The main advantage of this solution is that developers know which parts should be changed while bumping the cairo compiler. Without those comments, it is tough to do it efficiently

**Comment by Arcticae** [comment]
> I get the intentions but I am not a big fan of this PR, imo it encourages developers to blind copy the blockifier code without understanding 😞.

Well i would be happy to upstream all of this into blockifier but that seems impossible ATM. Perhaps if we have clear definitions of what we would need, and what we're modifying internally we could one day consider upstreaming, and the comments would help with defining requirements as well.

**Review by Radinyn — APPROVED** [review]

**Review by war-in — APPROVED** [review]

**Pull request merged**